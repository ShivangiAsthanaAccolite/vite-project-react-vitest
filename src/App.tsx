import "./App.css";

import About from "./component/About";
import Home from "./component/Home";
import MyComponent from "./component/MyComponent";
import { UserGreeting } from "./component/UserGreeting";
import UserList from "./component/UserList";

function App() {
  return (
    <>
      <div>
        <h4>App Component</h4>
        <p> I am in App component</p>
        <Home />
        <About />
        <UserGreeting />
        <UserList />
        <MyComponent />
      </div>
    </>
  );
}

export default App;
