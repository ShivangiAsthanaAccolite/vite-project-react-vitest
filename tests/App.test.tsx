import { render, screen } from "@testing-library/react";

import App from "../src/App";

describe("App", () => {
  it("renders the App component", () => {
    render(<App />);

    screen.debug(); // prints out the jsx in the App component unto the command line
  });

  it("App component snapshot test", () => {
    const { asFragment } = render(<App />);
    // Create a snapshot of the component's output
    expect(asFragment()).toMatchSnapshot();
  });
});
