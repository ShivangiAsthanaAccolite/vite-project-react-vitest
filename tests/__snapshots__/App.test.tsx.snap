// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > App component snapshot test 1`] = `
<DocumentFragment>
  <div>
    <h4>
      App Component
    </h4>
    <p>
       I am in App component
    </p>
    <div>
      <h3>
        Home Component
      </h3>
      <p
        data-testid="para"
      >
        This is paragraph in Home Component
      </p>
    </div>
    <div>
      <h5>
        About Component
      </h5>
      <button
        data-testid="button"
        disabled=""
      >
        Click Me!
      </button>
      <div>
        <input
          data-testid="element-to-focus"
          type="text"
        />
      </div>
    </div>
    <div>
      Hello, Guest!
    </div>
    <div>
      Gender: NA
    </div>
    <div>
      Loading...
    </div>
    <div />
  </div>
</DocumentFragment>
`;
